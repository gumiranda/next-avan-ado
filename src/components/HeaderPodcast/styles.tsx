import styled from 'styled-components';

export const Wrapper = styled.header`
  background: var(--white);
  height: 10.4rem;
  display: flex;
  align-items: center;
  padding: 3rem 5rem;
  border-bottom: 1px solid var(--gray-100);
`;
export const Logo = styled.img``;
export const Paragraph = styled.p`
  font-size: 1.4rem;
  margin-left: 2rem;
  padding: 0.75rem 0 0.75rem 3rem;
  border-left: 1px solid var(--gray-100);
`;
export const SpanDate = styled.span`
  font-size: 1.4rem;
  margin-left: auto;
  text-transform: capitalize;
`;
