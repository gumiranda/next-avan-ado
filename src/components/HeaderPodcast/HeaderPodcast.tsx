import * as S from './styles';
import format from 'date-fns/format';
import ptBR from 'date-fns/locale/pt-BR';
const HeaderPodcast = () => {
  const currentDate = format(new Date(), 'EEEEEE, d MMMM', { locale: ptBR });
  return (
    <S.Wrapper>
      <S.Logo src="/logo.svg" alt="HeaderPodcast" />
      <S.Paragraph>O melhor est√° aqui</S.Paragraph>
      <S.SpanDate>{currentDate}</S.SpanDate>
    </S.Wrapper>
  );
};
export default HeaderPodcast;
