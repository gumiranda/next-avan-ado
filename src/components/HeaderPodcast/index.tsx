import HeaderPodcast from './HeaderPodcast';
export default HeaderPodcast;
