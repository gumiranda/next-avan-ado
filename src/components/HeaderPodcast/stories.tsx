import { Story, Meta } from '@storybook/react/types-6-0';

import HeaderPodcast from './HeaderPodcast';

export default {
  title: 'HeaderPodcast',
  component: HeaderPodcast
} as Meta;

export const Basic: Story = (args) => <HeaderPodcast {...args} />;
