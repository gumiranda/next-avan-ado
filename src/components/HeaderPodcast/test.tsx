import { render, screen } from '@testing-library/react';

import HeaderPodcast from './HeaderPodcast';

describe('<HeaderPodcast />', () => {
  it('should render the logo image', () => {
    const { container } = render(<HeaderPodcast />);
    expect(screen.getByRole('img', { name: /HeaderPodcast/i })).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render the paragraph corretly', () => {
    const { container } = render(<HeaderPodcast />);
    expect(screen.getByText(/O melhor estÃ¡ aqui/i)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
  it('should render the span date corretly', () => {
    const { container } = render(<HeaderPodcast />);
    expect(screen.getByText(/Qui, 8 de abril/i)).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
