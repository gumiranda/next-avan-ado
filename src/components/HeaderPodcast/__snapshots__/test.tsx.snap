// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderPodcast /> should render the logo image 1`] = `
.c0 {
  background: var(--white);
  height: 10.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 3rem 5rem;
  border-bottom: 1px solid var(--gray-100);
}

.c1 {
  margin-left: 2rem;
  padding: 0.75rem 0 0.75rem 3rem;
  border-left: 1px solid var(--gray-100);
}

.c2 {
  margin-left: auto;
  text-transform: capitalize;
}

<header
  class="c0"
>
  <img
    alt="HeaderPodcast"
    class=""
    src="/logo.svg"
  />
  <p
    class="c1"
  >
    O melhor está aqui
  </p>
  <span
    class="c2"
  >
    sab, 24 abril
  </span>
</header>
`;

exports[`<HeaderPodcast /> should render the paragraph corretly 1`] = `
.c0 {
  background: var(--white);
  height: 10.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 3rem 5rem;
  border-bottom: 1px solid var(--gray-100);
}

.c1 {
  margin-left: 2rem;
  padding: 0.75rem 0 0.75rem 3rem;
  border-left: 1px solid var(--gray-100);
}

.c2 {
  margin-left: auto;
  text-transform: capitalize;
}

<header
  class="c0"
>
  <img
    alt="HeaderPodcast"
    class=""
    src="/logo.svg"
  />
  <p
    class="c1"
  >
    O melhor está aqui
  </p>
  <span
    class="c2"
  >
    sab, 24 abril
  </span>
</header>
`;

exports[`<HeaderPodcast /> should render the span date corretly 1`] = `
.c0 {
  background: var(--white);
  height: 10.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 3rem 5rem;
  border-bottom: 1px solid var(--gray-100);
}

.c1 {
  margin-left: 2rem;
  padding: 0.75rem 0 0.75rem 3rem;
  border-left: 1px solid var(--gray-100);
}

.c2 {
  margin-left: auto;
  text-transform: capitalize;
}

<header
  class="c0"
>
  <img
    alt="HeaderPodcast"
    class=""
    src="/logo.svg"
  />
  <p
    class="c1"
  >
    O melhor está aqui
  </p>
  <span
    class="c2"
  >
    sab, 24 abril
  </span>
</header>
`;
